tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/SelectEnvironmentProviderViewModel.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.Mvvm.Messaging;
+using DevHome.Common.Contracts.Services;
+using DevHome.Common.Environments.Models;
+using DevHome.SetupFlow.Models.Environments;
+using DevHome.SetupFlow.Services;
+using Serilog;
+using WinUIEx;
+
+namespace DevHome.SetupFlow.ViewModels.Environments;
+
+public partial class SelectEnvironmentProviderViewModel : SetupPageViewModelBase
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(SelectEnvironmentProviderViewModel));
+
+    private readonly WindowEx _windowEx;
+
+    private readonly IComputeSystemService _computeSystemService;
+
+    public ComputeSystemProviderDetails SelectedProvider { get; private set; }
+
+    [ObservableProperty]
+    private bool _areProvidersLoaded;
+
+    [ObservableProperty]
+    private int _selectedProviderIndex;
+
+    [ObservableProperty]
+    private ObservableCollection<ComputeSystemProviderViewModel> _providersViewModels;
+
+    public SelectEnvironmentProviderViewModel(
+        ISetupFlowStringResource stringResource,
+        SetupFlowOrchestrator orchestrator,
+        IComputeSystemService computeSystemService,
+        WindowEx windowEx)
+           : base(stringResource, orchestrator)
+    {
+        PageTitle = stringResource.GetLocalized(StringResourceKey.SelectEnvironmentPageTitle);
+        _computeSystemService = computeSystemService;
+        _windowEx = windowEx;
+    }
+
+    public async Task LoadProvidersAsync()
+    {
+        AreProvidersLoaded = false;
+        Orchestrator.NotifyNavigationCanExecuteChanged();
+
+        await Task.Run(async () =>
+        {
+            var providerDetails = await _computeSystemService.GetComputeSystemProvidersAsync();
+
+            _windowEx.DispatcherQueue.TryEnqueue(() =>
+            {
+                ProvidersViewModels = new();
+
+                foreach (var providerDetail in providerDetails)
+                {
+                    ProvidersViewModels.Add(new ComputeSystemProviderViewModel(providerDetail));
+                }
+
+                AreProvidersLoaded = true;
+            });
+        });
```

Comment:
This can be simplified and we can remove the window dependency.

```suggestion
        var providerDetails = await Task.Run(async () =>await _computeSystemService.GetComputeSystemProvidersAsync());
        ProvidersViewModels = new();

        foreach (var providerDetail in providerDetails)
        {
                ProvidersViewModels.Add(new ComputeSystemProviderViewModel(providerDetail));
        }

        AreProvidersLoaded = true;
```